### DISCUSSION TITLE
Add ResourceKey to ResourcesChangedEventArgs

### DISCUSSION THREAD

**Discussion by rabbitism** [AvaloniaUI/Avalonia#16208] [Ideas]
Currently when subscribing to IResourceHost.ResourceChanged event, we cannot know exactly which resource has been changed. Is it possible to add resource Key to this event args, just like PropertyChangedEventArgs?

**Comment by maxkatz6** [comment]
Most of the time it will be raised without resource key at all. I.e. when dictionary is initially attached, or owner changed. Basically with any operation that isn't simple `Add(key,value)`/`Remove(key)`. Either way it might be useful.

**Reply by rabbitism** [reply]
Yes, I think it's still a valid case to raise an arg without key, but currently, `GetResourceObservable(object key)` is not working at all, it reacts to all resource changes, instead of the particular key.
